desc/coils.py

```diff
@@ -1498,6 +1505,99 @@ def to_SplineXYZ(self, knots=None, grid=None, method="cubic", name=""):
         coils = [coil.to_SplineXYZ(knots, grid, method) for coil in self]
         return self.__class__(*coils, NFP=self.NFP, sym=self.sym, name=name)
 
+    def is_self_intersecting(self, grid=None, tol=None):
```

Comment:
where is this function used? I see it's in the test, but is it used anywhere in the coils code?

Reply:
`from_symmetry` of `CoilSet`

Reply:
It's actually used in `CoilSet.__init__`